Mathlib/Topology/ContinuousFunction/FunctionalCalculus.lean

```diff
@@ -0,0 +1,676 @@
+  refine cfcSpec_eq_of_continuous_of_map_id (cfcSpec_predicate ha f) φ ?_ ?_
+  · exact (cfcSpec_closedEmbedding ha).continuous.comp f'.continuous_comp_left
+  · simp only [StarAlgHom.comp_apply, ContinuousMap.compStarAlgHom'_apply]
+    congr
+    ext x
+    simp [hff']
+
+end cfcSpec
+
+section CFC
+
+-- right now these tactics are just wrappers, but potentially they could be more sophisticated.
+
+/-- A tactic used to automatically discharge goals relating to the continuous functional calculus,
+specifically whether the element satisfies the predicate. -/
+syntax (name := cfcTac) "cfc_tac" : tactic
+macro_rules
+  | `(tactic| cfc_tac) => `(tactic| (try (first | assumption | infer_instance | aesop)))
+
+-- if `fun_prop` is good enough, we'll just use that everywhere instead of this, but right now
+-- there are still a few rough edges. See #10724
+/-- A tactic used to automatically discharge goals relating to the continuous functional calculus,
+specifically concerning continuity of the functions involved. -/
+syntax (name := cfcContTac) "cfc_cont_tac" : tactic
+macro_rules
+  | `(tactic| cfc_cont_tac) => `(tactic| try (first | fun_prop (disch := aesop) | assumption))
+
+open Classical in
+/-- This is the *continuous functional calculus* of an element `a : A` applied to bare functions.
+When either `a` does not satisfy the predicate `p` (i.e., `a` is not `IsStarNormal`,
+`IsSelfAdjoint`, or `0 ≤ a` when `R` is `ℂ`, `ℝ`, or `ℝ≥0`, respectively), or when `f : R → R` is
+not continuous on the spectrum of `a`, then `cfc a f` returns the junk value `0`.
+
+This is the primary declaration intended for widespread use of the continuous functional calculus,
+and all the API applies to this declaration. For more information, see the module documentation
+for `Topology.ContinuousFunction.FunctionalCalculus`. -/
+noncomputable irreducible_def cfc (a : A) (f : R → R) : A :=
+  if h : p a ∧ ContinuousOn f (spectrum R a)
+    then cfcSpec h.1 ⟨_, h.2.restrict⟩
+    else 0
+
+variable (a : A)
+
+lemma cfc_apply (f : R → R) (ha : p a := by cfc_tac)
+    (hf : ContinuousOn f (spectrum R a) := by cfc_cont_tac) :
+    cfc a f = cfcSpec (a := a) ha ⟨_, hf.restrict⟩ := by
+  rw [cfc_def, dif_pos ⟨ha, hf⟩]
+
+lemma cfc_apply_of_not_and {f : R → R} (ha : ¬ (p a ∧ ContinuousOn f (spectrum R a))) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg ha]
+
+lemma cfc_apply_of_not_predicate {f : R → R} (ha : ¬ p a) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg (not_and_of_not_left _ ha)]
+
+lemma cfc_apply_of_not_continuous {f : R → R} (hf : ¬ ContinuousOn f (spectrum R a)) :
+    cfc a f = 0 := by
+  rw [cfc_def, dif_neg (not_and_of_not_right _ hf)]
+
+variable (R) in
+lemma cfc_id (ha : p a := by cfc_tac) : cfc a (id : R → R) = a :=
+  cfc_apply a (id : R → R) ▸ cfcSpec_map_id (p := p) ha
+
+variable (R) in
+lemma cfc_id' (ha : p a := by cfc_tac) : cfc a (fun x : R ↦ x) = a :=
+  cfc_id R a
```

Comment:
Can one of these be simp?

Reply:
It would fire if it were, see [lean4#3475](https://github.com/leanprover/lean4/issues/3475)